api/lang/completion_test.go

```diff
@@ -98,28 +101,204 @@ func TestCompletion(t *testing.T) {
 				Label:      "Say <hello> to <gauge>",
 				Detail:     "Step",
 				Kind:       lsp.CIKFunction,
-				TextEdit:   lsp.TextEdit{Range: lsp.Range{}, NewText: `Say "${1:hello}" to "${0:gauge}"`},
+				TextEdit:   lsp.TextEdit{Range: lsp.Range{Start: position, End: position}, NewText: `Say "${1:hello}" to "${0:gauge}"`},
 				FilterText: "Say <hello> to <gauge>",
 			},
 			InsertTextFormat: snippet,
 		},
 	},
 	}
 	provider = &dummyCompletionProvider{}
-	b, _ := json.Marshal(lsp.TextDocumentPositionParams{})
+
+	b, _ := json.Marshal(lsp.TextDocumentPositionParams{TextDocument: lsp.TextDocumentIdentifier{URI: "uri"}, Position: position})
 	p := json.RawMessage(b)
 
 	got, err := completion(&jsonrpc2.Request{Params: &p})
 
 	if err != nil {
-		t.Errorf("Expected error == nil in Completion, got %s", err.Error())
+		t.Fatalf("Expected error == nil in Completion, got %s", err.Error())
```

Comment:
Changed this mainly because Errorf does not fail the test now and exit. The next check present depends on the `err` not being nil. Fatalf does not move forward and fails the test completely at that point. Hence the change.